python/metatensor-torch/metatensor/torch/atomistic/model.py

```diff
@@ -45,6 +45,30 @@ def load_atomistic_model(path, extensions_directory=None) -> "MetatensorAtomisti
     return torch.jit.load(path)
 
 
+def is_atomistic_model(module: torch.nn.Module) -> bool:
+    """
+    Check if a loaded model is an metatensor atomistic model.
+
+    :param model: model to check
+    :raises TypeError: if the model is not a :py:class:`torch.nn.Module`
+    :return: ``True`` if the ``model`` has been exported, ``False`` otherwise.
+    """
+    if not isinstance(module, torch.nn.Module):
+        raise TypeError(
+            f"`module` should be a torch.nn.Module, not {type(module).__name__}"
+        )
```

Comment:
By making this check, you're assuming that the model is not "exported" (torchscripted), unless something has changed in torch.

IIRC, after torchscripting, a model becomes a `ScriptObject`